packages/headless/src/app/recommendation-engine/recommendation-engine-configuration.ts

```diff
@@ -40,6 +41,10 @@ export interface RecommendationEngineConfiguration extends EngineConfiguration {
    * America/Montreal
    */
   timezone?: string;
+  /**
+   * Allows for augmenting a search response before the state is updated.
+   */
+  preprocessSearchResponseMiddleware?: PostprocessSearchResponseMiddleware;
```

Comment:
I know the names are confusing with 'preprocess' and 'postprocess' but I simply followed what was already written for the search engine.